pyccel/codegen/printing/fcode.py

```diff
@@ -963,6 +963,9 @@ def _print_FunctionCallArgument(self, expr):
             return '{}'.format(self._print(expr.value))
 
     def _print_Constant(self, expr):
+        if expr == math_constants['nan']:
+            errors.report("Can't print nan in Fortran",
+                    severity='error', symbol=expr)
         val = LiteralFloat(expr.value)
         return self._print(val)
```

Comment:
It appears that this method only works for floating-point real constants. Should we not use `DtypePrecisionToCastFunction` instead, as is now done in the C printer?

Reply:
I think you are probably right but in #1811 this logic will be changed so that such constants are declared with `parameter`. I can change this if you really want but I'm not sure it's worth it given that it will not go in v1.12.0 and that it will be reverted almost immediately

Reply:
Until #1811 it is not possible to declare a `Constant` in the code to be translated so the use of `LiteralFloat` is sufficient for the current limits of the implementation

Reply:
If you are going to change this soon, it does not really matter.